### PR TITLE
Support JavaScript libraries via Asset Packagist

### PR DESCRIPTION
**Pull request by phenaproxima**
This will partially unblock acquia/lightning#425 and allow us to bring in any JavaScript asset library using the asset-packagist.org Composer repo.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,15 +21,19 @@
         "drupal/console": "^1.0"
     },
     "require": {
-        "composer/installers": "^1.0",
         "drupal-composer/drupal-scaffold": "^2.0.0",
         "cweagans/composer-patches": "^1.6.0",
-        "acquia/lightning": "~2.1.0"
+        "acquia/lightning": "~2.1.0",
+        "oomphinc/composer-installers-extender": "^1.1"
```
Comment (balsama) [inline composer.json]: Is this necessary? Is it a root requirement? Thinking we can get rid of it altogether since it's in Lightning.

Comment (phenaproxima) [inline composer.json]: We were directly requiring composer/installers before (which this supersedes), so as far as I can tell, it's a root requirement...we could open a follow-up PR to experiment with removing it, though.

Comment (balsama) [inline composer.json]: Agree. Let's fix it now though if we can. It changes the other PRs from four changes to two.

* ~~remove composer-installers~~ (optional)
* ~~Add composer-installers-extra~~
* Add packagist
* Add new path definitions

**Comment by balsama** [comment]
Perfect. If 1fa2071 passes, just change the lightning requirement back to ~2.1.6 and this is ready for merge. Also needs to be cherry picked to headless branch.

**Pull request merged**